addons/lunch/models/lunch_alert.py

```diff
@@ -84,36 +87,92 @@ def _search_available_today(self, operator, value):
             ])
         ])
 
-    def _notify_chat(self):
-        records = self.search([('mode', '=', 'chat'), ('active', '=', True)])
-
-        today = fields.Date.today()
-        now = fields.Datetime.now()
-
-        for alert in records:
-            notification_to = now.astimezone(pytz.timezone(alert.tz)).replace(second=0, microsecond=0, tzinfo=None)
-            notification_from = notification_to - timedelta(minutes=5)
-            send_at = datetime.combine(fields.Date.today(),
-                float_to_time(alert.notification_time, alert.notification_moment))
-
-            if alert.available_today and send_at > notification_from and send_at <= notification_to:
-                order_domain = [('state', '!=', 'cancelled')]
 
-                if alert.location_ids.ids:
-                    order_domain = expression.AND([order_domain, [('user_id.last_lunch_location_id', 'in', alert.location_ids.ids)]])
-
-                if alert.recipients != 'everyone':
-                    weeks = 1
+    @api.depends('name', 'mode', 'notification_time', 'notification_moment', 'tz', 'active')
+    def _compute_cron_id(self):
+        for alarm in self:
+            alarm = alarm.with_context(tz=alarm.tz)
+
+            cron_required = (
+                alarm.active
+                and alarm.mode == 'chat'
+                and (not alarm.until or fields.Date.context_today(alarm) <= alarm.until)
+            )
+
+            now_tz = fields.Datetime.context_timestamp(alarm, datetime.utcnow())
+            sendat_tz = datetime.combine(date.today(), float_to_time(alarm.notification_time, alarm.notification_moment, alarm.tz))
+            if sendat_tz < now_tz:  # notif sent today already
+                sendat_tz += timedelta(days=1)
+            sendat_utc = sendat_tz.astimezone(pytz.UTC).replace(tzinfo=None)
+
+            alarm.cron_id.name = f"Lunch: alarm chat notification ({alarm.name})"
```

Comment:
You clearly are abusing the philosophy of compute methods.
You are not computing the value of a field on a record, instead you modify another record.

And it contains some **non-trivial bug**, due to the way you abuse it.
Suppose you have created an alarm at 11:00; its corresponding cron is scheduled at 11:00.
Now go to the form, change the time to 12:00, and discard because the existing value is correct.
Nothing should have changed, because you discarded your changes without saving, right?
Then go check the cron: it is now scheduled at 12:00!